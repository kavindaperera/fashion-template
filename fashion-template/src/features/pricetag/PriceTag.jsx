import React from "react";

function PriceTag({ currency, price, discount, discountActive }) {

  return (
    <div data-testid="price-tag">
      { discountActive  &&  discount > 0 && price && (
          <p>
            <del style={{ color: "#565656", fontFamily: "Lato"  }}>
              {currency}
              {price.toFixed(2)}
            </del>
            <a style={{ color: "red", fontFamily: "Lato"  }}>
              {currency}
              { (price - (price * discount) / 100).toFixed(2)}
            </a>
          </p>
      )}

      { (!discountActive || discount == 0) && price && (
        <p style={{ color: "#565656", fontFamily: "Lato"  }}>
          {currency}
          {price.toFixed(2)}
        </p>
      )}
    </div>
  );
}

export default PriceTag;
