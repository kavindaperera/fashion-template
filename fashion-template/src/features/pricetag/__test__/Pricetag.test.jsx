import React from 'react';
import ReactDOM from 'react-dom';
import PriceTag from "../PriceTag";
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer'


afterEach(cleanup);

it("renders without crashing", ()=>{
    const div = document.createElement("div");
    ReactDOM.render(<PriceTag></PriceTag>,div)
});


it("renders button correctly",()=>{
    const {getByTestId} = render(<PriceTag currency='$' price={100} discountActive={true} discount="50"></PriceTag>)
    expect(getByTestId('price-tag')).toHaveTextContent("$100$50")
});

it("renders button correctly",()=>{
    const {getByTestId} = render(<PriceTag currency='$' price={100} discountActive={true} discount="0"></PriceTag>)
    expect(getByTestId('price-tag')).toHaveTextContent("$100")
});

it('matches snapshot 1', ()=>{
    const tree = renderer.create(<PriceTag currency='$' price={100} discountActive={true} discount="50"></PriceTag>).toJSON();
    expect(tree).toMatchSnapshot();
})

it('matches snapshot 2', ()=>{
    const tree = renderer.create(<PriceTag currency='$' price={100} discountActive={true} discount="0"></PriceTag>).toJSON();
    expect(tree).toMatchSnapshot();
})
