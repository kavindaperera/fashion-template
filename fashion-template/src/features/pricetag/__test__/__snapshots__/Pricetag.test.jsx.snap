// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  data-testid="price-tag"
>
  <p>
    <del
      style={
        Object {
          "color": "#565656",
          "fontFamily": "Lato",
        }
      }
    >
      $
      100
    </del>
    <a
      style={
        Object {
          "color": "red",
          "fontFamily": "Lato",
        }
      }
    >
      $
      50
    </a>
  </p>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="price-tag"
>
  <p
    style={
      Object {
        "color": "#565656",
        "fontFamily": "Lato",
      }
    }
  >
    $
    100
  </p>
</div>
`;

exports[`matches snapshot 3 1`] = `
<div
  data-testid="price-tag-cart"
>
  <p>
    <del
      style={
        Object {
          "color": "grey",
          "fontFamily": "Lato",
        }
      }
    >
      $
      100
    </del>
    <a
      style={
        Object {
          "color": "red",
          "fontFamily": "Lato",
        }
      }
    >
      $
      50
    </a>
  </p>
</div>
`;

exports[`matches snapshot 4 1`] = `
<div
  data-testid="price-tag-cart"
>
  <p
    style={
      Object {
        "color": "grey",
        "fontFamily": "Lato",
      }
    }
  >
    $
    100
  </p>
</div>
`;
